### ISSUE TITLE
Bug report: Cantata crashes when the mpd playlist changes rapidly

### ISSUE THREAD

**Issue by kalsan** [CDrummond/cantata#1269]
Calling `mpc clear; mpc add ...; mpc play` 10x in one second for a very long playlist causes cantata to crash:

`  /usr/lib/libc.so.6 : ()+0x368f0
  cantata : Song::setKey(int)+0x2b3
  cantata : MPDParseUtils::parseSong(QList<QByteArray> const&, MPDParseUtils::Location)+0x544
  cantata : MPDParseUtils::parseSongs(QByteArray const&, MPDParseUtils::Location)+0xf2
  cantata : MPDConnection::playListInfo()+0xc3
  cantata : MPDConnection::playListChanges()+0x56
  cantata : MPDConnection::parseIdleReturn(QByteArray const&)+0x3af
  cantata : MPDConnection::idleDataReady()+0xf5
  cantata : ()+0x3bb6ae
  /usr/lib/libQt5Core.so.5 : QObject::event(QEvent*)+0xe2
  /usr/lib/libQt5Widgets.so.5 : QApplicationPrivate::notify_helper(QObject*, QEvent*)+0x9c
  /usr/lib/libQt5Widgets.so.5 : QApplication::notify(QObject*, QEvent*)+0x2a6
  /usr/lib/libQt5Core.so.5 : QCoreApplication::notifyInternal2(QObject*, QEvent*)+0x120
  /usr/lib/libQt5Core.so.5 : QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*)+0x1f6
  /usr/lib/libQt5Core.so.5 : ()+0x2f9c44
`

**Comment by CDrummond** [comment]
No crash here with `for in in 'seq 0 9'; do mpc clear; mpc load Test; mpc play; done` where `Test` has 16000 tracks.

But, really - why? I'm 100% certain Cantata has more bugs, but why even test this? This is not a realistic scenario.

**Comment by kalsan** [comment]
Hahaha after reading the bug report does sound strange indeed :-D Here's why I came across the bug:

I have numerous keyboard shortcuts for playing my favourite music collections. When I press Ctrl+Meta+Shift+B, the following commands are performed:

- `mpc clear`
- `mpc random on`
- `mpc add a_directory_that_contains_2000_songs`
- `mpc play`

In order to test this shortcut, I had cantata open in order to see if they actually perform the work. I pressed the shortcut several times and cantata crashed. So no I did not write a script to test that behavior ;-) The crash is reproducible, but sometimes it occurs after 3x and sometiems after 10x pressing the shortcut.

I'm using version 2.3.0
